include/npcomp/Dialect/ATen/ATenOpInterface.td

```diff
@@ -0,0 +1,57 @@
+//===- ATenOpInterface.td ----------------------------------*- tablegen -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+include "mlir/IR/OpBase.td"
+
+#ifndef ATEN_OP_INTERFACES
+#define ATEN_OP_INTERFACES
+
+def StatisticsOpInterface : OpInterface<"StatisticsOpInterface"> {
+  let description = [{
+     This interface allows ops to expose a static operation profile,
+	  describing the computational behavior of their function.
+  }];
+
+  let methods = [
+    InterfaceMethod<
+      "Return statistics about the compute requirements of an op",
+      "std::map<std::string, uint64_t>", "getStatistics"
+    >,
+
+    InterfaceMethod<"Return memory transfer requirements of an operand",
+      "uint64_t", "getOperandTransferVolume", (ins "unsigned int":$idx, "bool":$read), /*methodBody=*/[{}], [{
+        ConcreteOp *op = static_cast<ConcreteOp *>(this);
+        if (!read) return 0;
+        auto v = *op->getODSOperands(idx).begin();
```

Comment:
The interface does not mention anything about the indexing being about ODS operands, this looks unusual and error prone to me.

(also `front()` is more idiomatic than `*... .begin()` I think)


(same in the next method)